docs/source-2.0/aws/aws-endpoints.rst

```diff
@@ -0,0 +1,375 @@
+the ``aws`` partition and uses a non-standard global region in the ``aws-cn`` partition:
+
+.. code-block:: smithy
+
+    @nonRegionalizedEndpoints {
+        endpointPattern: "service_dnsSuffix",
+        partitionEndpointSpecialCases: [
+            {
+                partition: "aws",
+                endpoint: "myservice.global.amazonaws.com"
+            },
+            {
+                partition: "aws-cn",
+                region: "cn-north-1"
+            }
+        ]
+    }
+    service MyService {
+        version: "2020-04-02"
+    }
+
+-----------------------------------
+PartitionEndpointSpecialCase object
+-----------------------------------
+
+A PartitionEndpointSpecialCase object contains the following properties:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 30 60
+
+    * - Property name
+      - Type
+      - Description
+    * - partition
+      - ``string``
+      - **Required**. The partition to special case (example: "aws").
+    * - endpoint
+      - ``string``
+      - The special cased endpoint template.
+    * - region
+      - ``string``
+      - Override the ``defaultGlobalRegion`` used in this partition.
+
+.. smithy-trait:: aws.endpoints#dualStackOnlyEndpoints
+.. _aws.endpoints#dualStackOnlyEndpoints-trait:
+
+----------------------------------------------
+``aws.endpoints#dualStackOnlyEndpoints`` trait
+----------------------------------------------
+
+Summary
+    Indicates that a service has only dualStack endpoints and should not
+    have the ``useDualStackEndpoint`` endpoint parameter.
+
+Trait selector
+    ``service``
+Trait value
+    Annotation trait
+
+Adding the dualStackOnlyEndpoints to a service modifies the generation of endpoints from
+_aws.endpoints#standardRegionalEndpoints-trait or _aws.endpoints#nonRegionalizedEndpoints-trait
+and removes the ``useDualStackEndpoint`` parameter and defaults the behavior to dualstack for
+all partitions that support it.
+
+The following example specifies a service that uses standard regional endpoint patterns and
+is dualStack only:
+
+.. code-block:: smithy
+
+     @standardRegionalEndpoints
+     @dualStackOnlyEndpoints
+     service MyService {
+         version: "2020-04-02"
+     }
+
+.. smithy-trait:: aws.endpoints#rulesBasedEndpoints
+.. _aws.endpoints#rulesBasedEndpoints-trait:
+
+-------------------------------------------
+``aws.endpoints#rulesBasedEndpoints`` trait
+-------------------------------------------
+
+Summary
+    Indicates that a service has hand written endpoint rules.
+
+Trait selector
+    ``service``
+Trait value
+    Annotation trait
+
+Services marked with the ``rulesBasedEndpoints`` trait have hand written endpoint rules that
+extend or replace their standard generated endpoint rules.  This trait marks the presence
+of hand written rules, which may be added to the model by a transformer,
+but does not specify their behavior.
+
+The following example specifies a service that has standard regional endpoints extended with
```

Comment:
Is there a need to define replace vs extend? How does extending work functionally?

Reply:
Added a bit more detail.

Reply:
I'm still not quite sure what this is saying and it would be clear to me based on this if the implementation of this trait, other traits, or something else entirely is responsible for the extension functionality. Can there be conflicts and how are they resolved? Is there an order of precedence?